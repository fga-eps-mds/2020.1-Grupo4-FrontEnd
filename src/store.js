import { configureStore } from '@reduxjs/toolkit';
import usersReducer from './slices/usersSlice';
import tutorialSlice from './slices/tutorialSlice'

export default configureStore({
  reducer: {
    users: usersReducer,
    tutorial: tutorialSlice
  }
});