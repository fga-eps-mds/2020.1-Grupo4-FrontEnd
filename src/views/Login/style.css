#container{
    padding: 5%;
    background: linear-gradient(309.23deg, rgba(136, 56, 255, 0.95) 3.17%, rgba(233, 66, 255, 0.5) 114.71%);
}

.login-form {
    max-height: 600px;
    display: flex;
    flex-direction: column;
    justify-content: center;
    font-family: Overpass;
    background-color: white;
    border-radius: 10px;
    width: 428px;
    height: 500px;
    margin: auto;
    margin-top: 10vw;
    margin-bottom: 10vw;
}

.login-form h1 {
    font-family: Overpass;
    font-size: 48px;
    margin: 0;
    margin-left: 92.5px;
    padding-top: 44px;
    color: var(--title-color);
}


.login-form label form {
    margin-left: 92.5px;
    margin-top: 4vw;
    color:var(--title-color);
    font-size: 16px;
}
.login-form label form input{
    outline: none;
    width: 235px;
    display: block;
    margin-bottom: 20px;
    border-width: 0.2px;

    box-sizing: border-box;
    padding: 5px;
    border-radius: 10px;
}

.login-form label form .button{
    margin-left: 0;
    margin-right: 0;
    width: 235px;
}


.login-resources p, a{
    text-align: center;
    font-family: Overpass;
}
.login-resources p:first-child {
    margin-top: 3vw;
    color: var(--secondary-color);
}

.login-resources p a {
    color: var(--secondary-color);
    text-decoration: none;

}

.login-resources p a:hover {
    color: var(--hover-button);
}

@media(max-width: 800px) {
    .login-form {
        max-height: 400px;
        width: 80vw;
        height: 140vw;
    }
   .login-form label form {
    margin-left: 15vw;
   }
   .login-form label form input{
        width: 50vw;
        margin-right: auto;
   }

   .login-form label form .button{
        width: 50vw
    }
    .login-form h1 {
        margin: 0;
        text-align: center;
    }
    .login-resources p{
        margin: 0;
    }
}
