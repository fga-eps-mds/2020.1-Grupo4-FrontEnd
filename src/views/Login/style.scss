.login {
  min-height: 100vh;
  padding: 5rem 1rem;
  display: flex;
  flex-direction: column;
  align-items: center;
  background: linear-gradient(
    309.23deg,
    rgba(136, 56, 255, 0.95) 3.17%,
    rgba(233, 66, 255, 0.5) 114.71%
  );

  @media (max-width: 600px) {
    padding: 2rem 1rem;
  }

  &__body {
    max-width: 400px;
    background-color: #f9f9f9;
    border-radius: 10px;
    padding: 3rem 4rem;
    margin: 0 auto;
    box-shadow: 0px 4px 4px rgba(51, 51, 51, 0.25),
      0px 4px 4px rgba(51, 51, 51, 0.25), 0px 1px 1px rgba(51, 51, 51, 0.12);

    h1 {
      font-weight: 600;
    }

    @media (max-width: 600px) {
      padding: 2rem;
      margin-top: 8vh;
    }

    &--form {
      margin-top: 2rem;
      button {
        margin: 0 auto;
        margin-top: 1rem;
      }
    }

    .form-action {
      position: relative;
      display:  flex;
      align-items: center;
      flex-direction: column;
      margin-top: 1rem;
      text-align: center;

      @media(max-width: 600px) {
        width: 100%;
      }

      .loader {
        position: absolute;
        bottom: calc(-100% + 2rem);
      }

      button {
        width: 10rem;
        margin-bottom: 1.5rem;
      }
    }
  }

  &__resources {
    display: flex;
    flex-direction: column;
    align-items: center;
    margin-top: 2rem;

    a {
      color: var(--secondary-color);
      &:hover {
        text-decoration: underline;
      }
    }
  }
}
