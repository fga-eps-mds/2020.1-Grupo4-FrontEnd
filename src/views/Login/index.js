import React from 'react';
import Header from '../../components/Header';
import Footer from '../../components/Footer';
import Button from '../../components/Button';
import '../../index.css'
import './style.css';

function Login() {
    return (
        < >
            <div id="container">
                <div className="login-form">
                    <h1>Entrar</h1>
                    <label>
                        <form onSubmit={function (e) {
                            e.preventDefault()
                        }}>
                            e-mail<input value="name" type="text" name="email" placeholder="email@email.com" />
                        senha <input value="senha" type="password" name="password" />
                            <Button>LOGIN</Button>
                        </form>
                    </label>
                    <div className="login-resources">
                        <p>Recuperar senha</p>
                        <p>NÃ£o possui conta? <span>Cadastra-se</span></p>
                    </div>
                </div>
            </div>
        </>
    );
}


export default Login