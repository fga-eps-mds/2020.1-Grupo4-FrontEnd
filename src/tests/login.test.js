import { shallow } from 'enzyme';
import React from 'react';
import Login from '../views/Login';
import 'mutationobserver-shim';
import { store } from '../store'

global.MutationObserver = window.MutationObserver;

it('Expect to render Login Component', () => {
  expect(shallow(<Login store={store}></Login>).length).toEqual(1)
})

it('Expect to click button', () => {
  const wrapper = shallow(<Login store={store} />);
  wrapper.dive('Button').simulate('click')
})