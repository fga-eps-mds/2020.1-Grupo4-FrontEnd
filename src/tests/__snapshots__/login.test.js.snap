// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect to render Login Component 1`] = `
<Fragment>
  <div
    className="login"
  >
    <div
      className="login__form"
    >
      <h1>
        Entrar
      </h1>
      <label>
        <form
          onSubmit={[Function]}
        >
          <p>
            e-mail
          </p>
          <input
            name="email"
            placeholder="email@email.com"
            type="text"
          />
          <p>
            senha
          </p>
          <input
            name="password"
            type="password"
          />
          <Button
            small={true}
          >
            LOGIN
          </Button>
        </form>
      </label>
      <div
        className="login__resources"
      >
        <p>
          Recuperar senha
        </p>
        <p>
          NÃ£o possui conta? 
          <Link
            to="/cadastro"
          >
            Cadastre-se
          </Link>
        </p>
      </div>
    </div>
  </div>
</Fragment>
`;
